mdiff/mdiff_test.go

```diff
@@ -84,6 +84,24 @@ func TestDiff(t *testing.T) {
 	})
 }
 
+func TestRegression(t *testing.T) {
+	t.Run("#12", func(t *testing.T) {
+		const contextWindow = 3
+
+		// Produce a chunk with a longer overlap than the size of the context
+		// window.  Without the fix, this will trigger a panic in unification.
+		lhs := mstr.Lines("X\nY\nZ\n\na\nb\nc")
```

Comment:
suggestion, feel free to ignore: in similar tests, I've found it useful to have a test helper like:

`func lines(lns ...string) []string { return lns }`

The inputs then look like `lines("X", "Y", "Z", "", "a", "b", "c")`, which I find easier to grok than the excape-ful version.

Wackier option: make mstr.Lines accept ...string, and behave as if each chunk is separated by a \n. IOW, aside from the empty input case, mstr.Lines(strs...) is shorthand for `strings.Split(strings.Join(strs, "\n"), "\n")`

Reply:
SGTM, done (re: `lines`). Having `mstr.Lines` accept multiple arguments is also an interesting idea, but I'll think about that separately. 🙂